appsec/src/extension/msgpack_helpers.c

```diff
@@ -14,9 +14,14 @@
 #include "php_helpers.h"
 #include "php_objects.h"
 
-static const int MAX_DEPTH = 32;
+static const size_t MAX_DEPTH_READING = 32;
+static const size_t MAX_DEPTH_WRITING = 20;
+static const size_t MAX_STR_LEN = 4096;
+static const size_t MAX_ARRAY_SIZE = 256;
+static THREAD_LOCAL_ON_ZTS bool data_truncated_ = false;
```

Comment:
I'm not sure if this approach is the correct. This will mark all waf.requests in a request as truncated, whereas it probably only happened for one of them. I think the truncation flag should be instead sent to the helper, so that for that specific waf run it could generate a metric with the tag (or not).